specification/network/resource-manager/Microsoft.Network/stable/2020-06-01/networkInterface.json

```diff
@@ -1139,6 +1139,10 @@
           "readOnly": true,
           "description": "A list of references to linked BareMetal resources."
         },
+        "dscpConfiguration": {
+          "$ref": "./dscpConfiguration.json#/definitions/DscpConfiguration",
```

Comment:
>DscpConfiguration [](start = 57, length = 17)

This will cause the payload of the entire dscpConfiguration resource to be included in the payload of the network interface resource. This is basically an anti-pattern for the following reasons:
- breaks RBAC because a user who has /read permission on a nic but not a dscp configuration should not be able to see the the dscp configuration
- break Policy an unamiguous alias for properties of dscpConfiguration does not exist
- breaks Azure Resource Graph and Change Tracking because a single change notification represents an update of 2 entities

The alternative is to have the `dscpConfiguration` property be a string that points to a resource ID of the dscpConfiguration resource or have it be an object with an id property. (NRP uses the latter pattern if I recall correctly in some places.)

Reply:
On second thought, it appears that your example is following the recommended pattern, but the schema definition implies the whole dscp configuration payload. Can you fix the schema to match the example? (NRP should have a common type that you can reuse for this already.)

---
In reply to: [451222161](https://github.com/Azure/azure-rest-api-specs/pull/9954#discussion_r451222161) [](ancestors = 451222161)

Reply:
Did the change so it is now using the sub resource reference rather than the full reference of it.